clients/src/main/java/org/apache/kafka/common/Cluster.java

```diff
@@ -240,8 +217,7 @@ public PartitionInfo partition(TopicPartition topicPartition) {
      * @return The number of partitions or null if there is no corresponding metadata
      */
     public Integer partitionCountForTopic(String topic) {
-        List<PartitionInfo> partitions = this.partitionsByTopic.get(topic);
-        return partitions == null ? null : partitions.size();
+        return partitionsByTopic.get(topic).size();
```

Comment:
Annoying as it is, I think we had better keep the current semantics and return `null` if the topic is not present.

Reply:
☹️